universe/ValueRefs.h

```diff
@@ -710,7 +695,21 @@ Statistic<T>::Statistic(std::unique_ptr<ValueRef<T>>&& value_ref, StatisticType
     m_stat_type(stat_type),
     m_sampling_condition(std::move(sampling_condition)),
     m_value_ref(std::move(value_ref))
-{}
+{
+    this->m_root_candidate_invariant = (!m_sampling_condition || m_sampling_condition->RootCandidateInvariant()) &&
```

Comment:
For clarification: you are leaving out the `Variable<T>::*Invariant` call because `Statistic<T>` initializes `Variable<T>` with `NON_OBJECT_REFERENCE`, right?

Reply:
Yes, the result `true` is already know and doesn't matter.